### PR TITLE
Config rework

### PR DESCRIPTION
**Pull request by KyleRAnderson**
## Summary of Changes
- Moved the `XboxValue` enum to its own class.
- Added critical initialization for Config (bug fix)
- Added more tests to config to make sure certian things that should work work.
- Moved some utility methods from the `FluidButton` class to the `XboxValue` enum where they belonged.
- Added some utility methods to the `XboxValue` enum.

## Testing Performed
**Environment**: Plyboy and Practice bot
While working on intake mechanisms I made this change and the config stuff worked.

Deployed to practice bot and made sure constants table appeared and updated when it should.

### CODE REVIEW

**Comment by KyleRAnderson** [comment]
@eandr127 this is ready for review.

File: src/test/java/ca/team2706/frc/robot/config/ConfigTest.java [hunk 1]
```diff
@@ -114,6 +118,60 @@ public void setUp() throws Exception {
         }
     }
 
+
+    private static class FakeFluidConstant extends FluidConstant<String> {
```
Comment (eandr127) [inline src/test/java/ca/team2706/frc/robot/config/ConfigTest.java]: Should this be a delegate instead?

Comment (KyleRAnderson) [reply src/test/java/ca/team2706/frc/robot/config/ConfigTest.java]: I don't see how I would make that into a Delegate. Do you mind providing an example of what you mean?

Comment (eandr127) [reply src/test/java/ca/team2706/frc/robot/config/ConfigTest.java]: Actually, I think the only way that a delegate could be used would be to mock FluidConstant. What you have is fine.

**Pull request merged**